### DISCUSSION TITLE
Add ability to add multiple auth options

### DISCUSSION THREAD

**Discussion by ghost** [Intevel/nuxt-directus#106] [Ideas]
I am using different auth options in the same nuxt app.
- For requesting files/images I am using the public role.
- For requesting data I am using a token.
- For previewing the content I am using a different token.

My suggestion:
Set the default auth option in `nuxt.config.ts` as until now.
Whenever we need a different auth option, we could simply overwrite the default config when using the composable, like:
```js
const { getThumbnail: publicThumbnail } = useDirectusFiles({ auth: false });
// or
const { getSingletonItem } = useDirectusItems({ auth: { token: 'a1b2c3d4e5f…' } });
// or
const { getItems } = useDirectusItems({ auth: { user, pass } });
```

I’ve looked at the repository’s code, but without refactoring the code it is currently not able to use different auth options …

**Comment by Intevel** [comment]
That's a great idea.
